exonum-java-binding/core/rust/src/runtime/java_service_runtime.rs

```diff
@@ -131,7 +132,26 @@ impl JavaServiceRuntime {
         let args =
             Self::build_jvm_arguments(jvm_config, runtime_config, service_config, internal_config)
                 .expect("Unable to build arguments for JVM");
-        jni::JavaVM::new(args).unwrap()
+
+        jni::JavaVM::new(args)
+            .map_err(Self::transform_jni_error)
+            .expect("Unable to create JVM")
+    }
+
+    /// Transforms JNI errors by converting JNI error codes of error of type `Other` to its string
+    /// representation.
+    fn transform_jni_error(error: Error) -> Error {
+        match error.0 {
+            ErrorKind::Other(code) => match code {
```

Comment:
I wonder if it makes sense to upstream anything :thinking: ?

Reply:
Potential improvement - enhance the `jni::errors::jni_error_code_to_result()` function by making it translate every JNI code to separate error (current behavior is kinda minimal implementation that translates `JNI_OK` and `thread detached` to Result::Ok and ErrorKind::ThreadDetached respectively, any other JNI error code -> ErrorKind::Other(error_code)). Pros: error messages come for free; Cons: it's obviously a breaking change for those who already rely on ErrorKind::Other(error_code).

Reply:
On a quick look, it seems to make sense! I'd submit an issue to jni-rs or our issue tracker.

Reply:
https://github.com/jni-rs/jni-rs/issues/166